src/methods/properties.js

```diff
@@ -1005,7 +1007,20 @@ export function OrdinaryGetOwnProperty(realm: Realm, O: ObjectValue, P: Property
   // 2. If O does not have an own property with key P, return undefined.
   let existingBinding = InternalGetPropertiesMap(O, P).get(InternalGetPropertiesKey(P));
   if (!existingBinding) {
-    if (O.isPartialObject() && !O.isSimpleObject()) {
+    if (O.isPartialObject()) {
+      invariant(realm.useAbstractInterpretation); // __makePartial will already have thrown an error if not
+      if (O.isSimpleObject()) {
+        if (P instanceof StringValue) P = P.value;
+        if (typeof P === "string") {
```

Comment:
We can do better here, but things get complicated and right now we don't have a motivating scenario.